l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -134,124 +129,150 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
+        _transferAllowanceToNTV(assetInfo, _amount, _prevMsgSender);
+        /* solhint-disable no-unused-vars */
+        bytes memory bridgeMintData = IL1StandardAsset(l1Asset).bridgeBurn{value: msg.value}({
+            _chainId: _chainId,
+            _mintValue: 0,
+            _assetInfo: assetInfo,
+            _prevMsgSender: _prevMsgSender,
+            _data: abi.encode(_amount, address(0))
+        });
 
-        if (!hyperbridgingEnabled[_chainId]) {
-            chainBalance[_chainId][_l1Token] += _amount;
-        }
         // Note that we don't save the deposited amount, as this is for the base token, which gets sent to the refundRecipient if the tx fails
-        emit BridgehubDepositBaseTokenInitiated(_chainId, _prevMsgSender, _l1Token, _amount);
+        emit BridgehubDepositBaseTokenInitiated(_chainId, _prevMsgSender, _assetInfo, _amount);
+        // We could use this data instead of the above
+        // emit BridgehubMintData(bridgeMintData);
+    }
+
+    /// @dev for backwards compatibility and to automatically register l1 assets, and we return the correct info.
+    function _getAssetProperties(bytes32 _assetInfo) internal returns (address l1Asset, bytes32 assetInfo) {
+        l1Asset = assetAddress[_assetInfo];
+        assetInfo = _assetInfo;
+        if (l1Asset == address(0) && (uint256(_assetInfo) <= type(uint160).max)) {
+            l1Asset = address(nativeTokenVault);
+            assetInfo = keccak256(abi.encode(block.chainid, NATIVE_TOKEN_VAULT_VIRTUAL_ADDRESS, _assetInfo));
+            nativeTokenVault.registerToken(address(uint160(uint256(_assetInfo))));
+        }
     }
 
-    /// @dev Transfers tokens from the depositor address to the smart contract address.
-    /// @return The difference between the contract balance before and after the transferring of funds.
-    function _depositFunds(address _from, IERC20 _token, uint256 _amount) internal returns (uint256) {
-        uint256 balanceBefore = _token.balanceOf(address(this));
-        // slither-disable-next-line arbitrary-send-erc20
-        _token.safeTransferFrom(_from, address(this), _amount);
-        uint256 balanceAfter = _token.balanceOf(address(this));
+    function decodeLegacyData(bytes calldata _data, address _prevMsgSender) external returns (bytes32, bytes memory) {
+        require(msg.sender == address(this));
+        (address _l1Token, uint256 _depositAmount, address _l2Receiver) = abi.decode(
+            _data,
+            (address, uint256, address)
+        );
+        bytes32 assetInfo = nativeTokenVault.getAssetInfo(_l1Token);
+        if (nativeTokenVault.tokenAddress(assetInfo) == address(0)) {
+            nativeTokenVault.registerToken(_l1Token);
+        }
+        _transferAllowanceToNTV(assetInfo, _depositAmount, _prevMsgSender);
+        return (assetInfo, abi.encode(_depositAmount, _l2Receiver));
+    }
+
+    function _transferAllowanceToNTV(bytes32 _assetInfo, uint256 _amount, address _prevMsgSender) internal {
```

Comment:
it does not transfer the allowance in case the token does not belong to NTV, I would suggest adding a comment about that or reflecting in name